### PR TITLE
feat: added types to Link component

### LINKED ISSUE #13561: Add TypeScript types to Link

### PR DESCRIPTION
**Pull request by SunnyJohal**
Closes #13561

Added typescript types to Link component.

#### Changelog

**New**

- Added typescript types to Link component.

**Changed**

-  File extensions from .js to .tsx

**Removed**

- n/a

#### Testing / Reviewing

- Run storybook, navigate to link component and check that it is still rendering.
- Run unit tests

### CODE REVIEW

File: packages/react/src/components/Link/Link.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    {
+      children,
+      className: customClassName,
+      href,
+      disabled = false,
+      inline = false,
+      visited = false,
+      renderIcon: Icon,
+      size,
+      target,
```
Comment (tay1orjones) [inline packages/react/src/components/Link/Link.tsx:81]: `target` needs to be added to propTypes

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Thanks! Just one small change needed

File: packages/react/src/components/Link/Link.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  AriaAttributes,
+  ComponentType,
+  HTMLAttributeAnchorTarget,
+  PropsWithChildren,
+} from 'react';
+import { usePrefix } from '../../internal/usePrefix';
+
+interface LinkProps {
+  /**
+   * @description Indicates the element that represents the
```
Comment (tay1orjones) [inline packages/react/src/components/Link/Link.tsx:21]: The `@description` JSDoc annotation is [not supported](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#unsupported-tags)

```suggestion
   * Indicates the element that represents the
```

Additional usages of `@description` in this PR will need updated as well

**Review by tay1orjones — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Link/Link.tsx (hunk 2 above)
Comment (SunnyJohal) [reply packages/react/src/components/Link/Link.tsx:21]: Will remove these now

File: packages/react/src/components/Link/Link.tsx (hunk 1 above)
Comment (SunnyJohal) [reply packages/react/src/components/Link/Link.tsx:81]: Will add this now

**Review by andreancardona — APPROVED** [review]
Thank you so much for this - looks good!

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**